### PR TITLE
handle completion commit + requests refactor

### PR DESCRIPTION
**Pull request by dimacodota**
handle completion commit + requests refactor

### CODE REVIEW

File: lib/requests.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            "net_length": net_length,
+            "strength": selected_completion.get("detail", ""),
+            "origin": selected_completion.get("origin", ""),
+            "index": completions.index(selected_completion),
+            "line_prefix_length": line_prefix_length,
+            "line_net_prefix_length": line_prefix_length - (length - net_length),
+            "line_suffix_length": current_line.end() - current_location,
+            "num_of_suggestions": len(completions),
+            "num_of_vanilla_suggestions": len(
+                [x for x in completions if x["origin"] == "VANILLA"]
```
Comment (boaz-codota) [inline lib/requests.py:71]: extract to an "enum" to group all the options of origin together

File: lib/requests.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    substitution,
+    selected_completion,
+    completions,
+):
+    line_prefix_length = (current_location - len(substitution)) - current_line.begin()
+    length = current_location - before_prefix_location
+    net_length = len(substitution)
+    request = {
+        "Selection": {
+            "language": [x for x in file_name.split(".")][-1],
```
Comment (boaz-codota) [inline lib/requests.py:60]: would throw an exception if there is no extension name...

File: lib/requests.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                [x for x in completions if x["origin"] == "CLOUD"]
+            ),
+            "num_of_lsp_suggestions": len(
+                [x for x in completions if x["origin"] == "LSP"]
+            ),
+            "suggestions": [
+                {
+                    "length": len(x["new_prefix"]),
+                    "strength": x.get("detail", ""),
+                    "origin": x.get("origin", ""),
```
Comment (boaz-codota) [inline lib/requests.py:86]: I think we send `UNKOWN` by default

**Review by boaz-codota — CHANGES_REQUESTED** [review]

File: lib/requests.py (hunk 2 above)
Comment (boaz-codota) [reply lib/requests.py:60]: it won't throw an exception, but wil make the data filled with file names. You need to send some default file extension if no extension exists

File: lib/requests.py (hunk 3 above)
Comment (dimacodota) [reply lib/requests.py:86]: done

File: lib/requests.py (hunk 2 above)
Comment (dimacodota) [reply lib/requests.py:60]: done, changed to `os.path.splitext` witch will return '' in case of no extension

File: lib/requests.py (hunk 1 above)
Comment (dimacodota) [reply lib/requests.py:71]: done

File: lib/requests.py [hunk 4]
```diff
@@ -57,36 +59,40 @@ def set_completion_state(
...
                 }
                 for x in completions
             ],
         }
     }
     set_state(request)
+
+
+def count_by_origin(completions, origin):
+    return len([x for x in completions if x["origin"] == origin])
```
Comment (boaz-codota) [inline lib/requests.py:98]: `len(...)` should work without a list (its probably minor I know, but it allocates an additional list)

File: lib/requests.py [hunk 5]
```diff
@@ -57,36 +59,40 @@ def set_completion_state(
     net_length = len(substitution)
     request = {
         "Selection": {
-            "language": [x for x in file_name.split(".")][-1],
+            "language": os.path.splitext(file_name)[1][1:],
```
Comment (boaz-codota) [inline lib/requests.py:62]: I have no idea what it does now (I prefer the previous one). I wanted you to supply a static file extension (something like `UNKNOWN` if there is no file extension.

**Review by boaz-codota — CHANGES_REQUESTED** [review]

**Review by boaz-codota — APPROVED** [review]

**Pull request merged**